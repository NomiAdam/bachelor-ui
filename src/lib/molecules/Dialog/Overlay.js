import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Container from '../../organisms/Container/index';

const StyledOverlayWrapper = styled(Container)`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 99;
  background-color: rgba(0,0,0,.3);
`;

const refuseScroll = (e) => {
    e.preventDefault();
    return false;
};

const Overlay = ({ handleClick, children }) => (
    <StyledOverlayWrapper height="100vh" onClick={ handleClick } onScroll={ refuseScroll }>
        { children }
    </StyledOverlayWrapper>
);

Overlay.propTypes = {
    /**
     * onClick function handler
     */
    handleClick: PropTypes.func,
    /**
     * Children nodes
     */
    children: PropTypes.any,
};

export default Overlay;
