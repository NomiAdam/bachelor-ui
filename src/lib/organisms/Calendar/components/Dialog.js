import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const minWidth = 200;

const StyledDialogWrapper = styled.div.attrs({
    style: ({
        left, top, width,
    }) => {
        const correctLeft = left > minWidth ? (left - minWidth - 15) : (left + width);
        return {
            top: `${ top - 10 }px`,
            left: `${ correctLeft }px`,
        };
    },
})`
  transition: top 1s, left 1s, opacity .5s, visibility .6s;
  transition-timing-function: ease;
  border: 1px solid black;
  opacity: ${ ({ isOpen }) => (isOpen ? '1' : '0') };
  visibility: ${ ({ isOpen }) => (isOpen ? 'visible' : 'hidden') };
  position: absolute;
  width: ${ minWidth }px;
  top: ${ ({ top }) => top }px;
  height: 250px; 
  z-index: 99;
`;

const Dialog = ({
    left, top, right, width, isOpen, elem: Elem,
}) => (
    <StyledDialogWrapper onClick={ e => e.stopPropagation() } right={ right } left={ left } top={ top } isOpen={ isOpen } width={ width }>
        <Elem />
    </StyledDialogWrapper>
);

Dialog.propTypes = {
    left: PropTypes.number,
    right: PropTypes.number,
    top: PropTypes.number,
    width: PropTypes.number,
    isOpen: PropTypes.bool,
    elem: PropTypes.node,
};

export default Dialog;
