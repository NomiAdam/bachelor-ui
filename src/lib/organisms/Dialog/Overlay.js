import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import flexConstant from '../../constants/flex';
import Container from '../../molecules/Container/index';

const StyledOverlayWrapper = styled(Container)`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 222;
  background-color: rgba(0,0,0,.3);
  display: ${ flexConstant.FLEX };
  justify-content: ${ flexConstant.POSITION_CENTER };
  align-items: ${ flexConstant.POSITION_CENTER };  
  overflow-y: auto;
`;

const refuseScroll = (e) => {
    e.preventDefault();
    return false;
};

const Overlay = ({ handleClick, children, secondary }) => (
    <StyledOverlayWrapper secondary={ secondary } height="100vh" onClick={ handleClick } onScroll={ refuseScroll }>
        { children }
    </StyledOverlayWrapper>
);

Overlay.propTypes = {
    /**
     * onClick function handler
     */
    handleClick: PropTypes.func,
    /**
     * Children nodes
     */
    children: PropTypes.any,
};

export default Overlay;
