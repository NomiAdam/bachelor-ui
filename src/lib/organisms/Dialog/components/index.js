import DialogActions from './DialogActions';
import DialogContent from './DialogContent';

export {
    DialogActions,
    DialogContent,
};
