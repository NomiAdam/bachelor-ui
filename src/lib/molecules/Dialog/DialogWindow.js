import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../organisms/Paper/index';
import { Grid, GridCol } from '../../organisms/Grid/index';
import { DialogContent, DialogActions } from './components/index';

const DialogWindow = ({
    actionChildren, children, heading, width, height, backgroundColor,
}) => (
    <Paper width={ width } height={ height } backgroundColor={ backgroundColor }>
        <Grid gridGap="0" padding="0">
            <GridCol colXS={ 12 }>
                {heading}
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogContent>
                    {children}
                </DialogContent>
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogActions>
                    { actionChildren }
                </DialogActions>
            </GridCol>
        </Grid>
    </Paper>
);

DialogWindow.propTypes = {
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    actionChildren: PropTypes.any.isRequired,
    /**
     * Node rendered on heading
     */
    heading: PropTypes.any.isRequired,
    /**
   * string definition of dialog width
   */
    width: PropTypes.string,
    /**
   * string definition of dialog height
   */
    height: PropTypes.string,
    /**
   * String representation of outer backgroundColor css property
   */
    backgroundColor: PropTypes.string,
};

export default DialogWindow;
