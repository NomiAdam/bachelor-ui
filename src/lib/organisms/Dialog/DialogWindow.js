import React from 'react';
import PropTypes from 'prop-types';
import Paper from '../../molecules/Paper/index';
import { Grid, GridCol } from '../../molecules/Grid/index';
import { DialogContent, DialogActions } from './components/index';

const preventClick = (e) => {
    e.stopPropagation();
    e.preventDefault();
    return false;
};

const DialogWindow = ({
    actionChildren, children, heading, width, height, backgroundColor,
}) => (
    <Paper width={ width } height={ height } backgroundColor={ backgroundColor } handleClick={ preventClick }>
        <Grid gridGap="0" padding="0">
            <GridCol colXS={ 12 }>
                {heading}
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogContent>
                    {children}
                </DialogContent>
            </GridCol>
            <GridCol colXS={ 12 }>
                <DialogActions>
                    { actionChildren }
                </DialogActions>
            </GridCol>
        </Grid>
    </Paper>
);

DialogWindow.propTypes = {
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    actionChildren: PropTypes.any.isRequired,
    /**
     * Node rendered on heading
     */
    heading: PropTypes.any.isRequired,
    /**
   * string definition of dialog width
   */
    width: PropTypes.string,
    /**
   * string definition of dialog height
   */
    height: PropTypes.string,
    /**
   * String representation of outer backgroundColor css property
   */
    backgroundColor: PropTypes.string,
};

export default DialogWindow;
