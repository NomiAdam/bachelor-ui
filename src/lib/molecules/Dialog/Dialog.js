import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { lightTheme } from '../../constants/theme';
import Overlay from '../../atoms/Overlay/index';
import DialogWindow from './DialogWindow';

class Dialog extends PureComponent {
    handleClose = () => {
        const { handleClose } = this.props;
        handleClose();
    };

    render() {
        const {
            isOpen, children, heading, renderActionChildren,
        } = this.props;
        return (
            <Overlay
                isOpen={ isOpen }
                handleClose={ this.handleClose }
                background="rgba(0, 0, 0, 0.2)"
                displayHeading={ false }
                backColor={ lightTheme.white }
            >
                <DialogWindow heading={ heading } renderActionChildren={ renderActionChildren }>
                    { children }
                </DialogWindow>
            </Overlay>
        );
    }
}

Dialog.propTypes = {
    /**
     * Boolean whether Dialog is open or not
     */
    isOpen: PropTypes.bool,
    /**
     * Function handler to handle onClose click
     */
    handleClose: PropTypes.func,
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    renderActionChildren: PropTypes.func.isRequired,
    /**
     * Heading node of dialog window
     */
    heading: PropTypes.any.isRequired,
};

export default Dialog;
