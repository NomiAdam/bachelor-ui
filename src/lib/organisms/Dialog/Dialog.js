import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { noop } from 'ramda-extension';
import DialogWindow from './DialogWindow';
import Overlay from './Overlay';

const StyledWrapper = styled.div`
  position: fixed;
  z-index: 111;
  display: ${ ({ isOpen }) => (isOpen ? 'flex' : 'none') };
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;  
`;

class Dialog extends PureComponent {
    render() {
        const {
            isOpen, children, heading, actionChildren, width, height, backgroundColor, secondary, closeOnOverlayClick,
            handleClose,
        } = this.props;
        return (
            <StyledWrapper isOpen={ isOpen }>
                <Overlay handleClick={ closeOnOverlayClick ? handleClose : noop }>
                    <DialogWindow
                        backgroundColor={ backgroundColor }
                        heading={ heading }
                        actionChildren={ actionChildren }
                        width={ width }
                        height={ height }
                        secondary={ secondary }
                    >
                        { children }
                    </DialogWindow>
                </Overlay>
            </StyledWrapper>
        );
    }
}

Dialog.propTypes = {
    /**
     * Boolean whether Dialog is open or not
     */
    isOpen: PropTypes.bool,
    /**
     * Children nodes
     */
    children: PropTypes.any,
    /**
     * Function handler to render action children
     */
    actionChildren: PropTypes.any,
    /**
     * Heading node of dialog window
     */
    heading: PropTypes.any,
    /**
   * Width of Dialog
   */
    width: PropTypes.string,
    /**
   * Height of Dialog
   */
    height: PropTypes.string,
    /**
   * Background color of inner wrapping element
   */
    backgroundColor: PropTypes.string,
    /**
   * Function handler to handle close of the modal upon action
   */
    handleClose: PropTypes.func,
    /**
   * Whether we should close modal on overlay click
   */
    closeOnOverlayClick: PropTypes.bool,
};

export default Dialog;
