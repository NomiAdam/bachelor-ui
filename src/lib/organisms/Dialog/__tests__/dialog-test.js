import React from 'react';
import Dialog from '../Dialog';
import { shallow } from 'enzyme';

const mockChildren = () => <h1>Mock</h1>;

describe('Dialog', () => {
    it('should be defined', () => {
        expect(Dialog).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<Dialog renderActionChildren={mockChildren} />)).toMatchSnapshot();
    });
    it('should match the snapshot when open', () => {
        expect(shallow(<Dialog renderActionChildren={mockChildren} isOpen />)).toMatchSnapshot();
    });
    it('should match the snapshot with children', () => {
        expect(shallow(<Dialog renderActionChildren={mockChildren} isOpen>{ mockChildren() }</Dialog>)).toMatchSnapshot();
    });
});
