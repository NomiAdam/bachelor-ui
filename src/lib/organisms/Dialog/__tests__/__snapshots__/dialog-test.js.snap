// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should match the snapshot 1`] = `
<styled.div>
  <Overlay
    handleClick={[Function]}
  >
    <DialogWindow />
  </Overlay>
</styled.div>
`;

exports[`Dialog should match the snapshot when open 1`] = `
<styled.div
  isOpen={true}
>
  <Overlay
    handleClick={[Function]}
  >
    <DialogWindow />
  </Overlay>
</styled.div>
`;

exports[`Dialog should match the snapshot with children 1`] = `
<styled.div
  isOpen={true}
>
  <Overlay
    handleClick={[Function]}
  >
    <DialogWindow>
      <h1>
        Mock
      </h1>
    </DialogWindow>
  </Overlay>
</styled.div>
`;
